README.md
#Ussher File
This is a line from RStudio
<<<<<<< HEAD
This package includes data from Ussher's Annals of the World.
=======
Line added from GitHub
>>>>>>> c947baa81fb93b34c63c98ffa45d3251dfca9c31


NEWS.md
# ussher 1.10

* Added a `NEWS.md` file to track changes to the package.


R/usshraw.R
#' Raw original data directly drawn with fundamental cleaning from "Annals_djvu.txt" Archbishop James Ussher's chronology of ancient history known popularly as The Annals of the World (1658).
#'
#' "usshraw" file - Truncated listing of amalgamated data drawn from original chronological textual historic events are classified by index, year, epoch (or one of the 7 ancient "Ages of the World", Biblical source book if referenced (rarely), as well as alternate dating mechanisms, such as "Anno Mundi" (age of the world) or Julian Period. Additional file "ussh.full" includes variables that may be of further interest to historians, such as Southern Kingdom and Northern Kingdom discrepant dates, and the original amalgamated Dating mechanic used by Ussher in the original text. The raw data can also be called using "ussh.raw"
#'
#' @format A tibble with 5011 rows and 3 variables:
#' \describe{
#'   \item{Event}{chr Ussher's original historical paragraph, cleaned for text processing}
#'   \item{Epoch}{chr denoting which of the 7 ancient "Ages of the World" the event falls into}
#'   \item{Dating}{chr the original full-text string originally created by Ussher to identify, compare and harmonize the various dating methods in expressing the algorithms an historical analysis. All numeric date variables are drawn from this original date grouping}
#' }
#' @source \url{https://archive.org/stream/AnnalsOfTheWorld/Annals_djvu.txt}
#' @examples
#' summary(usshraw)
"usshraw"



R/usshfull.R
#' Expanded data drawn from "ussher" Archbishop James Ussher's chronology of ancient history known popularly as The Annals of the World (1658).
#'
#' "usshfull" file - Expanded historical chronological textual historic events are classified by index, year, epoch (or one of the 7 ancient "Ages of the World", Biblical source book if referenced (rarely), as well as alternate dating mechanisms, such as "Anno Mundi" (age of the world) or Julian Period. Additional file "ussh.full" includes variables that may be of further interest to historians, such as Southern Kingdom and Northern Kingdom discrepant dates, and the original amalgamated Dating mechanic used by Ussher in the original text. The raw data can also be called using "ussh.raw"
#'
#' @format A tibble with 5011 rows and 8 variables:
#' \describe{
#'   \item{Index}{dbl The indexed number of Ussher's paragraphs, in ascending chronological order, added by later editors.}
#'   \item{EventTxt}{chr Ussher's original historical paragraph, cleaned for text processing}
#'   \item{YearBCAD}{dbl Harmonized year BC or AD proposed by Ussher. BC is a negative number. AD is positive.}
#'   \item{Epoch}{chr denoting which of the 7 ancient "Ages of the World" the event falls into}
#'   \item{BibBk1}{chr a Biblical Book reference, if one is included in the paragraph}
#'   \item{AnnoMund}{dbl denoting "Anno Mundus" or age of the world from Creation, or 4004 BC}
#'   \item{Season}{chr denoting which quarter of the year - Autumn, Winter, Spring or Summer that event was calculated to occur}
#'   \item{SKing}{chr denoting the year of the respective Southern Kingdom reign of the sitting king. This is a chr file because it is easily confused for a chronological date. Convert to numeric and associate with sitting king if necessary for more advanced date comparisons and algorithms}
#'   \item{NKing}{chr denoting the year of the respective Northern Kingdom reign of the sitting king. This is a chr file because it is easily confused for a chronological date. Convert to numeric and associate with sitting king if necessary for more advanced date comparisons and algorithms}
#'   \item{JulPer}{dbl denoting Julian Period date. Julian Period was backdated to creation by Ussher, following its adoption by Caesar in 43 BC}
#'   \item{Dating}{chr the original full-text string originally created by Ussher to identify, compare and harmonize the various dating methods in expressing the algorithms an historical analysis. All numeric date variables are drawn from this original date grouping}
#' }
#' @source \url{https://archive.org/stream/AnnalsOfTheWorld/Annals_djvu.txt}
#' @examples
#' summary(usshfull)
"usshfull"


data-raw/ussher_load.R
library(tidyverse)
library(stringr)
library(tidytext)
library(tidyr)
library(dplyr)

annals.data_raw <-
  read.csv("~/RStudio Files/packageussh/ussher/ussher/data-raw/UssherTextPreclean4.csv", stringsAsFactors = FALSE, skip = 11,sep=";")
annals.data_raw <- as_tibble(annals.data_raw)
annals.data.na1 <- annals.data_raw %>% dplyr::na_if("AgeBlank") %>% dplyr::na_if("DateBlank")
annals.data.fill1 <- annals.data.na1 %>% fill(Epoch) %>% fill(Dating)
annals.index <- annals.data.fill1 %>%
  mutate(Index = as.numeric(str_extract(Event, "[0-9]+\\."))) %>%
  mutate(TextSrc1 = as.character(str_extract_all(Event, "[A-Z][a-z]+ [0-9]+:[0-9]++-[0-9]+"))) %>%
  mutate(BibBk1 = as.character(str_extract(TextSrc1, "[A-Z][a-z]+"))) %>%
  mutate(Year = as.character(str_extract(Dating, "\\d{1,4}\\s*([B][C]|[A][D])")))%>%
  mutate(AnnoMund = as.character(str_extract(Dating, "\\d{1,4}[a-z]*\\s[A][M]")))%>%
  mutate(Season = as.character(str_extract(AnnoMund,"[a-z]")))%>%
  mutate(SKing = as.character(str_extract(Dating,"\\d{1,2}\\s[S][K]")))%>%
  mutate(NKing = as.character(str_extract(Dating,"\\d{1,2}\\s[N][K]"))) %>%
  mutate(JulPer = as.character(str_extract(Dating,"\\d{1,4}\\s[J][P]")))
annals.index <- annals.index %>%
  relocate(Year,.before = Event) %>%
  relocate(Dating,.after=JulPer)

ussh.ind <- annals.index %>%
  mutate(EventTxt=(str_replace_all(Event, "[A-Z][a-z]+ [0-9]+:[0-9]++-[0-9]"," "))) %>%
  mutate(EventTxt=(str_replace_all(EventTxt, "[[:punct:]]"," "))) %>%
  mutate(EventTxt=(str_replace_all(EventTxt, "[0-9]+"," "))) %>%
  mutate(EventTxt=(str_trim(EventTxt))) %>%
  mutate(Dating=(str_trim(Dating))) %>%
  mutate(AnnoMund=(str_remove_all(AnnoMund,"[a-z]"))) %>%
  mutate(AnnoMund=(str_remove_all(AnnoMund,"[A][M]"))) %>%
  mutate(EventTxt=(str_remove_all(EventTxt,"AgeBlank"))) %>%
  mutate(EventTxt=(str_remove_all(EventTxt,"ageblank"))) %>%
  mutate(EventTxt=(str_remove_all(EventTxt,"DateBlank"))) %>%
  mutate(EventTxt=(str_remove_all(EventTxt,"dateblank"))) %>%
  mutate(JulPer=(str_remove_all(JulPer,"[J][P]"))) %>%
  mutate(Season=(str_replace_all(Season,"[a]","Autumn"))) %>%
  mutate(Season=(str_replace_all(Season,"[b]","Winter"))) %>%
  mutate(Season=(str_replace_all(Season,"[c]","Spring"))) %>%
  mutate(Season=(str_replace_all(Season,"[d]","Summer"))) %>%
  mutate(Epoch=(str_replace_all(Epoch,"The Seventh Age of the World","7th Age"))) %>%
  mutate(Epoch=(str_replace_all(Epoch,"The Sixth Age of the World","6th Age"))) %>%
  mutate(Epoch=(str_replace_all(Epoch,"The Fifth Age of the World","5th Age"))) %>%
  mutate(Epoch=(str_replace_all(Epoch,"The Fourth Age of the World","4th Age"))) %>%
  mutate(Epoch=(str_replace_all(Epoch,"The Third Age of the World","3rd Age"))) %>%
  mutate(Epoch=(str_replace_all(Epoch,"The Second Age of the World","2nd Age"))) %>%
  mutate(Epoch=(str_replace_all(Epoch,"The First Age of the World","1st Age")))
ussh.ind$JulPer <-gsub(" $","",ussh.ind$JulPer,perl=T)
ussh.ind$AnnoMund <-gsub(" $","",ussh.ind$AnnoMund,perl=T)

ussh.ind <- select(ussh.ind, -Event)
ussh.ind <- ussh.ind  %>%
  relocate(EventTxt) %>%
  relocate(Index,.before=EventTxt)
ussh.ind <- ussh.ind  %>%
  mutate(BCnum=as.character(str_extract_all(Year,"\\d{1,4}\\s*[B][C]")))%>%
  mutate(ADnum=as.character(str_extract_all(Year,"\\d{1,4}\\s*[A][D]"))) %>%
  mutate(BCnum=(str_remove_all(BCnum,"[B][C]"))) %>%
  mutate(ADnum=(str_remove_all(ADnum,"[A][D]")))
ussh.ind[ussh.ind == "character(0)"] <- NA
ussh.ind$BCnum <-gsub(" $","",ussh.ind$BCnum,perl=T)
ussh.ind$ADnum <-gsub(" $","",ussh.ind$ADnum,perl=T)
ussh.ind$BCnum <- as.numeric(as.character(ussh.ind$BCnum))*-1
ussh.ind$ADnum <- as.numeric(as.character(ussh.ind$ADnum))
ussh.ind$AnnoMund <- as.numeric(as.character(ussh.ind$AnnoMund))
ussh.ind$JulPer <- as.numeric(as.character(ussh.ind$JulPer))
ussh.ind <- ussh.ind %>%
  mutate_at(vars(c("BCnum","ADnum")), ~replace_na(.,0))%>%
  rowwise() %>%
  mutate(BCAD = sum(c_across(BCnum:ADnum)))
ussh.ind <-select(ussh.ind, -c(Year, BCnum, ADnum))
ussh.ind <- ussh.ind %>%
  rename(YearBCAD = BCAD) %>%
  relocate(YearBCAD,.after = EventTxt)
usshraw <- annals.data_raw
usshfull <-ussh.ind
ussh.ind <- select(ussh.ind, -c(TextSrc1,SKing,NKing,Dating))

ussher <- ussh.ind
usethis::use_data(usshraw,overwrite=TRUE)
usethis::use_data(usshfull,overwrite=TRUE)
usethis::use_data(ussher, overwrite = TRUE)


R/ussher.R
#' Cleaned and tidied data drawn from Archbishop James Ussher's chronology of ancient history known popularly as The Annals of the World (1658).
#'
#' "ussher" file - Chronological textual historic events are classified by index, year, epoch (or one of the 7 ancient "Ages of the World", Biblical source book if referenced (rarely), as well as alternate dating mechanisms, such as "Anno Mundi" (age of the world) or Julian Period. Additional file "usshfull" includes variables that may be of further interest to historians, such as Southern Kingdom and Northern Kingdom discrepant dates, and the original amalgamated Dating mechanic used by Ussher in the original text. The raw data can also be called using "usshraw"
#'
#' @format A tibble with 5011 rows and 8 variables:
#' \describe{
#'   \item{Index}{dbl The indexed number of Ussher's paragraphs, in ascending chronological order, added by later editors.}
#'   \item{EventTxt}{chr Ussher's original historical paragraph, cleaned for text processing}
#'   \item{YearBCAD}{dbl Harmonized year BC or AD proposed by Ussher. BC is a negative number. AD is positive.}
#'   \item{Epoch}{chr denoting which of the 7 ancient "Ages of the World" the event falls into}
#'   \item{BibBk1}{chr a Biblical Book reference, if one is included in the paragraph}
#'   \item{AnnoMund}{dbl denoting "Anno Mundus" or age of the world from Creation, or 4004 BC}
#'   \item{Season}{chr denoting which quarter of the year - Autumn, Winter, Spring or Summer that event was calculated to occur}
#'   \item{JulPer}{dbl denoting Julian Period date. Julian Period was backdated to creation by Ussher, following its adoption by Caesar in 43 BC}
#' }
#' @source \url{https://archive.org/stream/AnnalsOfTheWorld/Annals_djvu.txt}
#' @examples
#' summary(ussher)
"ussher"
